.github/workflows/main.yml

```diff
@@ -0,0 +1,49 @@
+name: Kmesh CI Workflow
+
+on:
+  push:
+    branches: [ "main"]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        go-version: [ '1.21' ]
+    
+    env:
+      GOPATH: /home/runner/go
+      GOBIN: /home/runner/go/bin
+      PKG_CONFIG_PATH: $GITHUB_WORKSPACE/mk
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup Go
+      uses: actions/setup-go@v4.0.0
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Install Go Dependence
+      run: |
+        go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.32
+        sudo cp $GOBIN/* /usr/bin
+    
+    - name: Build Kmesh
+      shell: bash
+      run: |
+        sudo bash ./build.sh
+    
+    # Since requiring code generated by the Proto compilation, execute go lint after building.  
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v3.7.0
+      with:
+        args: "--out-format colored-line-number"
+        skip-pkg-cache: true
+
+    - name: Go Test
```

Comment:
Separate Unit test Job from Compile

Reply:
As mentioned in comments, UT and Lint for go require the compilation of proto.

Reply:
Provide some references,
- [cilium/ebpf](https://github.com/cilium/ebpf/blob/main/.github/workflows/ci.yml) gather compile and lint together
- [cilium/cilium](https://github.com/cilium/cilium/blob/main/.github/workflows/lint-go.yaml) uses a separate lint workflow.